index.js

```diff
@@ -11,13 +19,17 @@ module.exports = (fromDir, moduleId) => {
 
 	const fromFile = path.join(fromDir, 'noop.js');
 
-	try {
-		return Module._resolveFilename(moduleId, {
-			id: fromFile,
-			filename: fromFile,
-			paths: Module._nodeModulePaths(fromDir)
-		});
-	} catch (err) {
-		return null;
+	if (silent) {
+		try {
+			_resolveFileName(fromDir, fromFile, moduleId);
```

Comment:
Funny, the tests were covering it and it worked, there was another return statement at the end of the function, for the "non silent" case.
because it didn't fall to the `cache` it just worked :)